### DISCUSSION TITLE
How do you specify a path to the browser binary?

### DISCUSSION THREAD

**Discussion by joshuastegmaier** [ElSnoMan/pyleniumio#324] [Q&A]
I've found this issue: https://github.com/ElSnoMan/pyleniumio/issues/265, but the solution there does not appear to work, as it generates `selenium.common.exceptions.InvalidArgumentException: Message: invalid argument: unrecognized capability: chrome.binary`

I'm attempting to use Chrome installed via the @puppeteer/browser npm package, but I can't find a way to do so with pylenium (except possibly by modifying my PATH.)

**Comment by ElSnoMan** [comment]
Where does puppeteer install it?

**Reply by joshuastegmaier** [reply]
It installs it in a cache directory inside the working directory, at least by default. Inside that, it's put into a directory based on the version of Chrome.

I've got working for now by modifying my PATH temporary. I'm using Pylenium along with Django's testing framework (which uses unittest.

Because puppeteer returns the version and path of the browser it installs, our testing code (run through Github Actions) those this to use puppeteer's Chrome (using the shell to parse out the actual path):

    npm install
    npx gulp build
    chromepath=$(npx @puppeteer/browsers install chrome@stable)
    chromepath=${chromepath#* }
    chromepath=${chromepath%/chrome}
    OLDPATH=$PATH
    PATH=$PATH:$chromepath
    pipenv run ./manage.py collectstatic --no-input
    pipenv run coverage run ./manage.py test
    PATH=$OLDPATH`

**Comment by ElSnoMan** [comment]
I'm glad you found a workaround! Your setup is definitely complicated, but
I don't know your context -- maybe all of this is required
